doc/examples/sample_report.py

```diff
@@ -8,8 +8,8 @@
 # fname = 'EMVA1288_ReferenceSet_003_Simulation_12Bit/EMVA1288_Data.txt'
 # fname = os.path.join(dir_, fname)
 
-dataset_generator = DatasetGenerator(width=640,
-                                     height=480,
+dataset_generator = DatasetGenerator(width=160,
```

Comment:
I understand that for tests we need smaller images so it generates fast.
I'm not fan of callable python files, but in this case, I would prefer that we add a `if __name__ == __main__` and use `arparse` to change the width and height for the tests. 
What do you think?

Reply:
just to clarify why I want to keep relatively large images.
When developing the report, smaller image give strange spatial graphics, making it harder to differentiate between report look from actual processing problem

Reply:
I think a clear and easy to understand example is a priority. If you think larger images are necessary, I would suggest to adjust the file via sed in the CI.

Reply:
I don't understand why using sed is easier than just creating an entry point?

Reply:
I have only a very rudimentary knowledge of Python and its concepts so far. But as far as I understand it, this variant would have to extend the Python script doc example by something that is not relevant in the context of the doc. If you look at it only from the point of view of a documentation reader, it is easier to leave out the entry point stuff. A `if __name__ == '__main__'` will throw up the question of why this is necessary at this point. In the implementation of the library, I would be okay with such an extension, but in the documentation I think it's best to focus entirely on the relevant content.

Overall, the solution with sed is of course a bit more complex, but this complexity is encapsulated in the CI script and thus in a place that hardly any user of the repo will come into contact with.

Reply:
I thought again about the `if __name__ == '__main__'` variant. We would call the actual script function in the `if __name__ == '__main__'` branch. In the CI we include the script via another file, which means that the `if __name__ == '__main__'` branch is not executed. Consequently, we can call the script function with its own parameters.

If we set up all examples this way, I think this is okay as "code overhead". Especially since this seems to be a common pattern in Python, and moving the code from the global namespace to that of a function has a few more advantages. (Have read a bit about this in the meantime).

Only downside would be that we wouldn't test the `if __name__ == '__main__'` branch within the CI, of course. So if we pass invalid parameters in this branch, users will get an error message when running the example script, but the CI won't get anything out of it.

Should I adjust the PR accordingly? (The `if __name__ == '__main__'` I only include in the report for now. The adaptation of the other examples would be done in a separate PR).

Reply:
@bebuch yes, let's adapt this one only